#!/bin/sh /usr/share/dpatch/dpatch-run
## 02-switch_to_genisoimage.dpatch by Didier Raboud <didier@raboud.com>
##
## DP: Switch kdenlive from mkisofs to genisoimage.
##     Closes: #547501

@DPATCH@

diff -Naur kdenlive-0.7.5.orig/src/dvdwizard.cpp kdenlive-0.7.5/src/dvdwizard.cpp
--- kdenlive-0.7.5.orig/src/dvdwizard.cpp	2009-09-21 11:57:50.000000000 +0200
+++ kdenlive-0.7.5/src/dvdwizard.cpp	2009-09-21 12:02:14.000000000 +0200
@@ -525,7 +525,7 @@
     m_mkiso->setProcessChannelMode(QProcess::MergedChannels);
     QListWidgetItem *isoitem =  m_status.job_progress->item(4);
     isoitem->setIcon(KIcon("system-run"));
-    m_mkiso->start("mkisofs", args);
+    m_mkiso->start("genisoimage", args);
 
 }
 
diff -Naur kdenlive-0.7.5.orig/src/dvdwizardvob.cpp kdenlive-0.7.5/src/dvdwizardvob.cpp
--- kdenlive-0.7.5.orig/src/dvdwizardvob.cpp	2009-09-21 11:57:50.000000000 +0200
+++ kdenlive-0.7.5/src/dvdwizardvob.cpp	2009-09-21 12:02:14.000000000 +0200
@@ -53,7 +53,7 @@
     m_view.vobs_list->setIconSize(QSize(60, 45));
 
     if (KStandardDirs::findExe("dvdauthor").isEmpty()) m_errorMessage.append(i18n("<strong>Program %1 is required for the DVD wizard.</strong>", i18n("dvdauthor")));
-    if (KStandardDirs::findExe("mkisofs").isEmpty()) m_errorMessage.append(i18n("<strong>Program %1 is required for the DVD wizard.</strong>", i18n("mkisofs")));
+    if (KStandardDirs::findExe("genisoimage").isEmpty()) m_errorMessage.append(i18n("<strong>Program %1 is required for the DVD wizard.</strong>", i18n("genisoimage")));
     if (m_errorMessage.isEmpty()) m_view.error_message->setVisible(false);
     else m_view.error_message->setText(m_errorMessage);
 
diff -Naur kdenlive-0.7.5.orig/src/wizard.cpp kdenlive-0.7.5/src/wizard.cpp
--- kdenlive-0.7.5.orig/src/wizard.cpp	2009-09-21 11:57:50.000000000 +0200
+++ kdenlive-0.7.5/src/wizard.cpp	2009-09-21 12:02:14.000000000 +0200
@@ -337,7 +337,7 @@
     item = new QTreeWidgetItem(m_check.programList, QStringList() << QString() << i18n("Mkisofs"));
     item->setData(1, Qt::UserRole, i18n("Required for creation of DVD ISO images"));
     item->setSizeHint(0, itemSize);
-    if (KStandardDirs::findExe("mkisofs").isEmpty()) item->setIcon(0, m_badIcon);
+    if (KStandardDirs::findExe("genisoimage").isEmpty()) item->setIcon(0, m_badIcon);
     else item->setIcon(0, m_okIcon);
 
     // set up some default applications
